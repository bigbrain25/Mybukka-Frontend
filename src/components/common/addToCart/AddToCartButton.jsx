/* eslint-disable array-callback-return */
/* eslint-disable react/prop-types */
import React from 'react';

import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Column from 'Components/grid/Column';
import Row from 'Components/grid/Row';
import Button from 'Components/button/Button';

import './addToCartButton.scss';

const AddToCartButton = ({
  price,
  handleClick,
  itemIsInCart,
  submenus,
}) => {
  const handleTotal = () => {
    let total = 0;
    const result = submenus.handleSubmenu(submenus.submenus, submenus.customerSubmenu);
    if (result.length > 0) {
      result.map(subMenu => subMenu.options.map((option) => {
        total += option.price || 0;
      }));
    }
    return total;
  };

  return (
    <Column classNames="col-12 col-sm-12 col-xs-12 col-md-12 col-lg-8">
      <Button
        handleClick={itemIsInCart ? () => ({}) : handleClick}
        classNames={
          itemIsInCart
            ? 'disabled-primary-button add-to-cart-button'
            : 'primary-button add-to-cart-button'
        }
        type="button"
        dataDismiss={itemIsInCart ? '' : 'modal'}
      >
        <Row>
          <Column classNames="col-6">
            <span className="add-to-cart-text">ADD TO CART</span>
          </Column>
          <Column classNames="col-6">
            <span className="price-cart-button">â‚¦{handleTotal() + price}.00</span>
          </Column>
        </Row>
      </Button>
    </Column>
  );
};

const mapStateToProps = ({
  fetchBukkaMenuReducer: { mealToDisplay },
  // cartReducer: { items }
}) => ({
  mealToDisplay,
  // items,
  // itemIsInCart:
  // items.filter(item => item.slug === mealToDisplay.slug).length > 0
});

export default connect(
  mapStateToProps,
  null
)(AddToCartButton);

AddToCartButton.propTypes = {
  price: PropTypes.number.isRequired,
  handleClick: PropTypes.func.isRequired
};
