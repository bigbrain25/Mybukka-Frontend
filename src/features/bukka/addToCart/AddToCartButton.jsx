import React from 'react';
import PropTypes from 'prop-types';

import Column from 'Components/grid/Column';
import Row from 'Components/grid/Row';
import Button from 'Components/button/Button';

import './addToCartButton.scss';

const AddToCartButton = ({ price, handleClick }) => (
  <Column classNames="col-12 col-sm-12 col-xs-12 col-md-12 col-lg-8">
    <Button
      handleClick={handleClick}
      classNames="primary-button add-to-cart-button"
      type="button"
      dataDismiss="modal"
    >
      <Row>
        <Column classNames="col-6">
          <span className="add-to-cart-text">ADD TO CART</span>
        </Column>
        <Column classNames="col-6">
          <span className="price-cart-button">â‚¦{price}.00</span>
        </Column>
      </Row>
    </Button>
  </Column>
);

export default AddToCartButton;

AddToCartButton.propTypes = {
  price: PropTypes.number.isRequired,
  handleClick: PropTypes.func.isRequired
};
