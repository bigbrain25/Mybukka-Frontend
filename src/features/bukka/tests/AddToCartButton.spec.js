import React from 'react';
import AddToCartButton from '../addToCart/AddToCartButton';

describe('Add To Cart Button', () => {
  const props = {
    handleClick: jest.fn(),
    price: 15,
  };
  const wrapper = mount(<AddToCartButton {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('triggers the handleClick function when button is called', () => {
    const button = wrapper.find('.add-to-cart-button');
    button.simulate('click');

    expect(props.handleClick).toBeCalled();
  });
});
