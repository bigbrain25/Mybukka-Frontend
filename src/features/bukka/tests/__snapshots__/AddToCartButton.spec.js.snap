// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add To Cart Button renders properly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <AddToCartButton
    handleClick={[MockFunction]}
    price={15}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "handleClick": [MockFunction],
      "price": 15,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Button
          classNames="primary-button add-to-cart-button"
          dataTarget=""
          dataToggle=""
          disabled={false}
          handleClick={[MockFunction]}
          id="F-izl4Sdq"
          text=""
          type="button"
        >
          <Row
            classNames=""
          >
            <Column
              classNames="col-6"
            >
              <span
                className="add-to-cart-text"
              >
                ADD TO CART
              </span>
            </Column>
            <Column
              classNames="col-6"
            >
              <span
                className="price-cart-button"
              >
                ₦
                15
                .00
              </span>
            </Column>
          </Row>
        </Button>,
        "classNames": "col-12 col-sm-12 col-xs-12 col-md-12 col-lg-8",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="col col-12 col-sm-12 col-xs-12 col-md-12 col-lg-8"
        >
          <button
            class="primary-button add-to-cart-button"
            data-target=""
            data-testid="button"
            data-toggle=""
            id="F-izl4Sdq"
            type="button"
          >
            <div
              class="row "
            >
              <div
                class="col col-6"
              >
                <span
                  class="add-to-cart-text"
                >
                  ADD TO CART
                </span>
              </div>
              <div
                class="col col-6"
              >
                <span
                  class="price-cart-button"
                >
                  ₦
                  15
                  .00
                </span>
              </div>
            </div>
          </button>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            classNames="primary-button add-to-cart-button"
            dataTarget=""
            dataToggle=""
            disabled={false}
            handleClick={[MockFunction]}
            id="F-izl4Sdq"
            text=""
            type="button"
          >
            <Row
              classNames=""
            >
              <Column
                classNames="col-6"
              >
                <span
                  className="add-to-cart-text"
                >
                  ADD TO CART
                </span>
              </Column>
              <Column
                classNames="col-6"
              >
                <span
                  className="price-cart-button"
                >
                  ₦
                  15
                  .00
                </span>
              </Column>
            </Row>
          </Button>,
          "className": "col col-12 col-sm-12 col-xs-12 col-md-12 col-lg-8",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Row
                classNames=""
              >
                <Column
                  classNames="col-6"
                >
                  <span
                    className="add-to-cart-text"
                  >
                    ADD TO CART
                  </span>
                </Column>
                <Column
                  classNames="col-6"
                >
                  <span
                    className="price-cart-button"
                  >
                    ₦
                    15
                    .00
                  </span>
                </Column>
              </Row>,
              "classNames": "primary-button add-to-cart-button",
              "dataTarget": "",
              "dataToggle": "",
              "disabled": false,
              "handleClick": [MockFunction],
              "id": "F-izl4Sdq",
              "text": "",
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": <button
                class="primary-button add-to-cart-button"
                data-target=""
                data-testid="button"
                data-toggle=""
                id="F-izl4Sdq"
                type="button"
              >
                <div
                  class="row "
                >
                  <div
                    class="col col-6"
                  >
                    <span
                      class="add-to-cart-text"
                    >
                      ADD TO CART
                    </span>
                  </div>
                  <div
                    class="col col-6"
                  >
                    <span
                      class="price-cart-button"
                    >
                      ₦
                      15
                      .00
                    </span>
                  </div>
                </div>
              </button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Row
                  classNames=""
                >
                  <Column
                    classNames="col-6"
                  >
                    <span
                      className="add-to-cart-text"
                    >
                      ADD TO CART
                    </span>
                  </Column>
                  <Column
                    classNames="col-6"
                  >
                    <span
                      className="price-cart-button"
                    >
                      ₦
                      15
                      .00
                    </span>
                  </Column>
                </Row>,
                "className": "primary-button add-to-cart-button",
                "data-target": "",
                "data-testid": "button",
                "data-toggle": "",
                "disabled": false,
                "id": "F-izl4Sdq",
                "onClick": [MockFunction],
                "type": "button",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <Column
                        classNames="col-6"
                      >
                        <span
                          className="add-to-cart-text"
                        >
                          ADD TO CART
                        </span>
                      </Column>,
                      <Column
                        classNames="col-6"
                      >
                        <span
                          className="price-cart-button"
                        >
                          ₦
                          15
                          .00
                        </span>
                      </Column>,
                    ],
                    "classNames": "",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": <div
                      class="row "
                    >
                      <div
                        class="col col-6"
                      >
                        <span
                          class="add-to-cart-text"
                        >
                          ADD TO CART
                        </span>
                      </div>
                      <div
                        class="col col-6"
                      >
                        <span
                          class="price-cart-button"
                        >
                          ₦
                          15
                          .00
                        </span>
                      </div>
                    </div>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <Column
                          classNames="col-6"
                        >
                          <span
                            className="add-to-cart-text"
                          >
                            ADD TO CART
                          </span>
                        </Column>,
                        <Column
                          classNames="col-6"
                        >
                          <span
                            className="price-cart-button"
                          >
                            ₦
                            15
                            .00
                          </span>
                        </Column>,
                      ],
                      "className": "row ",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "children": <span
                            className="add-to-cart-text"
                          >
                            ADD TO CART
                          </span>,
                          "classNames": "col-6",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": <div
                            class="col col-6"
                          >
                            <span
                              class="add-to-cart-text"
                            >
                              ADD TO CART
                            </span>
                          </div>,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": <span
                              className="add-to-cart-text"
                            >
                              ADD TO CART
                            </span>,
                            "className": "col col-6",
                          },
                          "ref": null,
                          "rendered": Array [
                            Object {
                              "instance": <span
                                class="add-to-cart-text"
                              >
                                ADD TO CART
                              </span>,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "children": "ADD TO CART",
                                "className": "add-to-cart-text",
                              },
                              "ref": null,
                              "rendered": Array [
                                "ADD TO CART",
                              ],
                              "type": "span",
                            },
                          ],
                          "type": "div",
                        },
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "children": <span
                            className="price-cart-button"
                          >
                            ₦
                            15
                            .00
                          </span>,
                          "classNames": "col-6",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": <div
                            class="col col-6"
                          >
                            <span
                              class="price-cart-button"
                            >
                              ₦
                              15
                              .00
                            </span>
                          </div>,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": <span
                              className="price-cart-button"
                            >
                              ₦
                              15
                              .00
                            </span>,
                            "className": "col col-6",
                          },
                          "ref": null,
                          "rendered": Array [
                            Object {
                              "instance": <span
                                class="price-cart-button"
                              >
                                ₦
                                15
                                .00
                              </span>,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "children": Array [
                                  "₦",
                                  15,
                                  ".00",
                                ],
                                "className": "price-cart-button",
                              },
                              "ref": null,
                              "rendered": Array [
                                "₦",
                                "15",
                                ".00",
                              ],
                              "type": "span",
                            },
                          ],
                          "type": "div",
                        },
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                  "type": [Function],
                },
              ],
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "handleClick": [MockFunction],
        "price": 15,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Button
            classNames="primary-button add-to-cart-button"
            dataTarget=""
            dataToggle=""
            disabled={false}
            handleClick={[MockFunction]}
            id="F-izl4Sdq"
            text=""
            type="button"
          >
            <Row
              classNames=""
            >
              <Column
                classNames="col-6"
              >
                <span
                  className="add-to-cart-text"
                >
                  ADD TO CART
                </span>
              </Column>
              <Column
                classNames="col-6"
              >
                <span
                  className="price-cart-button"
                >
                  ₦
                  15
                  .00
                </span>
              </Column>
            </Row>
          </Button>,
          "classNames": "col-12 col-sm-12 col-xs-12 col-md-12 col-lg-8",
        },
        "ref": null,
        "rendered": Object {
          "instance": <div
            class="col col-12 col-sm-12 col-xs-12 col-md-12 col-lg-8"
          >
            <button
              class="primary-button add-to-cart-button"
              data-target=""
              data-testid="button"
              data-toggle=""
              id="F-izl4Sdq"
              type="button"
            >
              <div
                class="row "
              >
                <div
                  class="col col-6"
                >
                  <span
                    class="add-to-cart-text"
                  >
                    ADD TO CART
                  </span>
                </div>
                <div
                  class="col col-6"
                >
                  <span
                    class="price-cart-button"
                  >
                    ₦
                    15
                    .00
                  </span>
                </div>
              </div>
            </button>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              classNames="primary-button add-to-cart-button"
              dataTarget=""
              dataToggle=""
              disabled={false}
              handleClick={[MockFunction]}
              id="F-izl4Sdq"
              text=""
              type="button"
            >
              <Row
                classNames=""
              >
                <Column
                  classNames="col-6"
                >
                  <span
                    className="add-to-cart-text"
                  >
                    ADD TO CART
                  </span>
                </Column>
                <Column
                  classNames="col-6"
                >
                  <span
                    className="price-cart-button"
                  >
                    ₦
                    15
                    .00
                  </span>
                </Column>
              </Row>
            </Button>,
            "className": "col col-12 col-sm-12 col-xs-12 col-md-12 col-lg-8",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Row
                  classNames=""
                >
                  <Column
                    classNames="col-6"
                  >
                    <span
                      className="add-to-cart-text"
                    >
                      ADD TO CART
                    </span>
                  </Column>
                  <Column
                    classNames="col-6"
                  >
                    <span
                      className="price-cart-button"
                    >
                      ₦
                      15
                      .00
                    </span>
                  </Column>
                </Row>,
                "classNames": "primary-button add-to-cart-button",
                "dataTarget": "",
                "dataToggle": "",
                "disabled": false,
                "handleClick": [MockFunction],
                "id": "F-izl4Sdq",
                "text": "",
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": <button
                  class="primary-button add-to-cart-button"
                  data-target=""
                  data-testid="button"
                  data-toggle=""
                  id="F-izl4Sdq"
                  type="button"
                >
                  <div
                    class="row "
                  >
                    <div
                      class="col col-6"
                    >
                      <span
                        class="add-to-cart-text"
                      >
                        ADD TO CART
                      </span>
                    </div>
                    <div
                      class="col col-6"
                    >
                      <span
                        class="price-cart-button"
                      >
                        ₦
                        15
                        .00
                      </span>
                    </div>
                  </div>
                </button>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Row
                    classNames=""
                  >
                    <Column
                      classNames="col-6"
                    >
                      <span
                        className="add-to-cart-text"
                      >
                        ADD TO CART
                      </span>
                    </Column>
                    <Column
                      classNames="col-6"
                    >
                      <span
                        className="price-cart-button"
                      >
                        ₦
                        15
                        .00
                      </span>
                    </Column>
                  </Row>,
                  "className": "primary-button add-to-cart-button",
                  "data-target": "",
                  "data-testid": "button",
                  "data-toggle": "",
                  "disabled": false,
                  "id": "F-izl4Sdq",
                  "onClick": [MockFunction],
                  "type": "button",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <Column
                          classNames="col-6"
                        >
                          <span
                            className="add-to-cart-text"
                          >
                            ADD TO CART
                          </span>
                        </Column>,
                        <Column
                          classNames="col-6"
                        >
                          <span
                            className="price-cart-button"
                          >
                            ₦
                            15
                            .00
                          </span>
                        </Column>,
                      ],
                      "classNames": "",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": <div
                        class="row "
                      >
                        <div
                          class="col col-6"
                        >
                          <span
                            class="add-to-cart-text"
                          >
                            ADD TO CART
                          </span>
                        </div>
                        <div
                          class="col col-6"
                        >
                          <span
                            class="price-cart-button"
                          >
                            ₦
                            15
                            .00
                          </span>
                        </div>
                      </div>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <Column
                            classNames="col-6"
                          >
                            <span
                              className="add-to-cart-text"
                            >
                              ADD TO CART
                            </span>
                          </Column>,
                          <Column
                            classNames="col-6"
                          >
                            <span
                              className="price-cart-button"
                            >
                              ₦
                              15
                              .00
                            </span>
                          </Column>,
                        ],
                        "className": "row ",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "function",
                          "props": Object {
                            "children": <span
                              className="add-to-cart-text"
                            >
                              ADD TO CART
                            </span>,
                            "classNames": "col-6",
                          },
                          "ref": null,
                          "rendered": Object {
                            "instance": <div
                              class="col col-6"
                            >
                              <span
                                class="add-to-cart-text"
                              >
                                ADD TO CART
                              </span>
                            </div>,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": <span
                                className="add-to-cart-text"
                              >
                                ADD TO CART
                              </span>,
                              "className": "col col-6",
                            },
                            "ref": null,
                            "rendered": Array [
                              Object {
                                "instance": <span
                                  class="add-to-cart-text"
                                >
                                  ADD TO CART
                                </span>,
                                "key": undefined,
                                "nodeType": "host",
                                "props": Object {
                                  "children": "ADD TO CART",
                                  "className": "add-to-cart-text",
                                },
                                "ref": null,
                                "rendered": Array [
                                  "ADD TO CART",
                                ],
                                "type": "span",
                              },
                            ],
                            "type": "div",
                          },
                          "type": [Function],
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "function",
                          "props": Object {
                            "children": <span
                              className="price-cart-button"
                            >
                              ₦
                              15
                              .00
                            </span>,
                            "classNames": "col-6",
                          },
                          "ref": null,
                          "rendered": Object {
                            "instance": <div
                              class="col col-6"
                            >
                              <span
                                class="price-cart-button"
                              >
                                ₦
                                15
                                .00
                              </span>
                            </div>,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": <span
                                className="price-cart-button"
                              >
                                ₦
                                15
                                .00
                              </span>,
                              "className": "col col-6",
                            },
                            "ref": null,
                            "rendered": Array [
                              Object {
                                "instance": <span
                                  class="price-cart-button"
                                >
                                  ₦
                                  15
                                  .00
                                </span>,
                                "key": undefined,
                                "nodeType": "host",
                                "props": Object {
                                  "children": Array [
                                    "₦",
                                    15,
                                    ".00",
                                  ],
                                  "className": "price-cart-button",
                                },
                                "ref": null,
                                "rendered": Array [
                                  "₦",
                                  "15",
                                  ".00",
                                ],
                                "type": "span",
                              },
                            ],
                            "type": "div",
                          },
                          "type": [Function],
                        },
                      ],
                      "type": "div",
                    },
                    "type": [Function],
                  },
                ],
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
